### PR TITLE
remove honcho in the dev environment (just use foregrounded supervisor)

### PR DESCRIPTION
**Pull request by ryanpetrello**
context: https://github.com/ansible/awx/issues/3846#issuecomment-490565355

using supervisor gives us the ability to restart entire processes on code change (like the dispatcher and callback receiver)

### CODE REVIEW

**Comment by ryanpetrello** [comment]
related: https://github.com/ansible/awx/issues/3846

**Comment by kdelee** [comment]
is `tools/docker-compose/Procfile` used by anything anymore or could it be removed?

**Comment by jbradberry** [comment]
> is `tools/docker-compose/Procfile` used by anything anymore or could it be removed?

No, I'm pretty sure from looking at the honcho docs earlier that it is specific to honcho.  Kill it.

File: Makefile [hunk 1]
```diff
@@ -289,7 +281,7 @@ uwsgi: collectstatic
 	@if [ "$(VENV_BASE)" ]; then \
 		. $(VENV_BASE)/awx/bin/activate; \
 	fi; \
-    uwsgi -b 32768 --socket 127.0.0.1:8050 --module=awx.wsgi:application --home=/venv/awx --chdir=/awx_devel/ --vacuum --processes=5 --harakiri=120 --master --no-orphans --py-autoreload 1 --max-requests=1000 --stats /tmp/stats.socket --lazy-apps --logformat "%(addr) %(method) %(uri) - %(proto) %(status)" --hook-accepting1="exec:awx-manage run_dispatcher --reload"
+    uwsgi -b 32768 --socket 127.0.0.1:8050 --module=awx.wsgi:application --home=/venv/awx --chdir=/awx_devel/ --vacuum --processes=5 --harakiri=120 --master --no-orphans --py-autoreload 1 --max-requests=1000 --stats /tmp/stats.socket --lazy-apps --logformat "%(addr) %(method) %(uri) - %(proto) %(status)" --hook-accepting1-once="exec:supervisorctl restart tower-processes:awx-dispatcher tower-processes:awx-receiver"
```
Comment (jbradberry) [inline Makefile]: Wait, are we back to `--hook-accepting1-once` again?  How is that working?

File: tools/docker-compose/supervisor.conf [hunk 2]
```diff
@@ -32,6 +32,10 @@ command = make uwsgi
 autostart = true
 autorestart = true
 redirect_stderr=true
+stopwaitsecs = 1
```
Comment (ryanpetrello) [inline tools/docker-compose/supervisor.conf:35]: Without this incantation, I couldn't get supervisor to *actually* reliably kill uwsgi when it exited 😢

Maybe some variant of https://github.com/Supervisor/supervisor/issues/1101 ?

Comment (jbradberry) [reply tools/docker-compose/supervisor.conf:35]: Weird.

**Review by jbradberry — APPROVED** [review]
I think it's good.  :moyai:

**Pull request merged**